src/parameter.mligo

```diff
@@ -0,0 +1,20 @@
+// SPDX-FileCopyrightText: 2021 TQ Tezos
+// SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+#if !PARAMETER_H
+#define PARAMETER_H
+#include "types.mligo"
+#include "implementation.mligo"
```

Comment:
This seems to be working, but taking into account my previous comment about `contract_extra` I suppose that it would result in a need for another `storage.mligo` file as a result.

I recommend doing something else instead: move all the types except `storage` and `parameter` in a base module, e.g. `types/base.mligo` and then then have `types.mligo` instead only contain:
1. the import of `types/base.mligo`
2. the import of `implementation.mligo`
3. the definition of `parameter` and `storage`.

In this way everything except the implementation module can import `types.mligo` and find everything there...
(The implementation modules can't, but that's ok as they shouldn't be allowed to access the full storage nor parameter anyway).

Note: the `decision_lambda_input` and `decision_lambda_output` wouldn't fit well in this, but I think they could be removed.

Reply:
>In this way everything except the implementation module can import types.mligo and find everything there...

But implementation module import modules that inturn depend on `types.mligo`. I thought it would somehow work, but it does not seem to.

Reply:
:thinking: I think that implementation modules only import helpers, common modules and such, AFAICT these could also import only a `types/base.mligo` module alone.

It takes away a bit of the convenience, but it might still be worth it.

Reply:
:thinking: AFAICT some of them do indeed, I missed that.

Not sure if we could change something about those, but I'd leave this for a separate issue.

Never mind this comment(s) then :ok_hand: